### ISSUE TITLE
Getting no POST body in the Jenkins job

### ISSUE THREAD

**Issue by Mo0rBy** [jenkinsci/generic-webhook-trigger-plugin#277]
Jenkins version: 2.412
Generic webhook trigger plugin version: 1.87.0

I had a more complicated configuration at first that didn't work. I've made it much simpler to debug and found the issue, but don't know how to solve it.

I am trying to trigger my Jenkins job using a webhook that is very similar to a Github PR webhook. Right now, I've stripped back my configuration so that I am just creating a `PAYLOAD` variable that is the entire JSON POST body, but it doesn't get through into the job script.

Here is my JobDSL configuration:
```
    triggers {
        genericTrigger {
            genericVariables {
                genericVariable {
                    key("PAYLOAD")
                    value("$")
                    expressionType("JSONPath") //Optional, defaults to JSONPath
                }
            }
            token('PR_merged')
            tokenCredentialId('')
            silentResponse(false)
            regexpFilterText('')
            regexpFilterExpression('')
        }
    }  
```

Here is the output of my job when I have the `Print post content` option enabled.
```
Generic Cause
GenericWebhookEnvironmentContributor
 Received:




Checking out git ssh://git@........

<Some other output here that is not important>

[Pipeline] // podTemplate
[Pipeline] End of Pipeline
Also:   org.jenkinsci.plugins.workflow.actions.ErrorAction$ErrorId: 6eeecf99-4cf8-47ea-8470-dabfc8e9b5a2
groovy.lang.MissingPropertyException: No such property: PAYLOAD for class: groovy.lang.Binding
	at groovy.lang.Binding.getVariable(Binding.java:63)
```

I am just trying to simply use the `PAYLOAD` variable and print it out within the job so I know it's coming through, it wasn't working, so I enabled the `Print post content` option, and as you can see there is no content being shown. The webhook content type is set to `application/json` and the webhook on the service that sends the webhook (and triggers the job) is showing that Jenkins is responding with a 200 response and it has the following response body:
```
{"jobs":{"TOOLS/HOOKS/job_hook_pr_merged":{"regexpFilterExpression":"","triggered":true,"resolvedVariables":{"PAYLOAD":""},"regexpFilterText":"","id":4230,"url":"queue/item/4230/"}},"message":"Triggered jobs."}
```

What have I done incorrectly here?

**Comment by tomasbjerre** [comment]
Can you trigger the plugin with a curl command? There are examples in the readme.

**Comment by Mo0rBy** [comment]
Amazing, I've solved it.

Did a curl command as you advised and got literally nothing back. Just went straight back into my terminal and didn't see any job triggering. This was using `http://`. Switched the url to `https://` and then I saw the 200 response in my terminal, saw the webhook trigger and saw the POST content body.
After this, I went back to my webhook settings on the triggering service and changed the url to `https://` and it worked! I didn't think that would be causing the issue as I have Istio setup such that all HTTP traffic is redirected to HTTPS.

Thank you!
